packages/safety-suite-api/src/api/models/types.ts

```diff
@@ -115,23 +115,281 @@ export type OrderBy = {
+   *
+   * Depth is set by `relationsLevel`.
+   */
   relationModels?: string;
   relationNames?: string[];
   relationTemplateIds?: Id[];
+  /**
+   * Depth of related models to retrieve.
+   *
+   * Example:
+   * ```ts
+   * // Output model with relationsLevel === 2
+   * {
+   *   // ...
+   *   relationModels: {
+   *     accident: {
+   *       // ...
+   *       relationModels: {
+   *         employee: {
+   *           // ...
+   *         }
+   *       }
+   *     }
+   *   }
+   * }
+   * ```
+   */
   relationsLevel?: number;
+  /**
+   * Array of names of computations with `computeOn === 'read'` to skip computation.
+   */
   restrictReadComputations?: string[];
+  /**
+   * Array of names of computations with `computeOn === 'read'` to force computation.
+   */
   updateReadComputations?: boolean;
+  /**
+   * Will include `createdByUser` in returned models.
+   */
   users?: boolean;
 };
 
 export type ModelOptions = ModelFilterOptions &
   ModelListOptions &
   ModelOutputOptions;
+
+/**
+ * Determines how `queryPath` will be used.
+ */
+export enum QueryPathType {
+  /**
+   * Used to match relation ids.
+   *
+   * Example:
+   * ```ts
+   * operation: {type: OperationType.equals},
+   * queryPath: {
+   *   pathType: QueryPathType.relation,
+   *   path: ['employee']
+   * },
+   * queryValue: {value: 1} // ID of an Employee model
+   * ```
+   */
+  relation = 'RELATION',
+  /**
+   * Used to match field or computation values.
+   *
+   * Example:
+   * ```ts
+   * queryPath: {
+   *   pathType: QueryPathType.path,
+   *   path: ['firstName']
+   * }
+   * ```
+   */
+  path = 'PATH',
+  group = 'GROUP',
+  template = 'TEMPLATE',
+  /**
+   * Used to match model ids.
+   */
+  id = 'ID',
+  /**
+   * Used to match fields in relations.
+   *
+   * Example:
+   * ```ts
+   * queryPath: {
+   *   pathType: QueryPathType.relationPath,
+   *   path: ['employee', 'firstName']
+   * }
+   * ```
+   */
+  relationPath = 'RELATION_PATH',
+  /**
+   * Used to match the name of the `createdBy` user.
+   */
+  userName = 'USER_NAME',
+  /**
+   * Used to match the integration source.
+   */
+  source = 'SOURCE'
+}
+/**
+ * Determines how the `conditions` will applied.
+ */
+export enum QueryNodeOperation {
+  /**
+   * Will return models that match all of the `conditions`.
+   */
+  and = 'AND',
+  /**
+   * Will return models that match any of the `conditions`.
+   */
+  or = 'OR',
+  /**
+   * Will ignore any `conditions`.
+   */
+  empty = 'EMPTY'
+}
+/**
+ * Determines how `queryValue.value` will be evaluated.
+ */
+export enum OperationType {
+  equals = 'EQUALS',
+  notEquals = 'NOT_EQUALS',
+  isDistinctFrom = 'IS_DISTINCT_FROM',
+  lessThan = 'LESS_THAN',
+  greaterThan = 'GREATER_THAN',
+  lessThanEqual = 'LESS_THAN_EQUAL',
+  greaterThanEqual = 'GREATER_THAN_EQUAL',
+  hasValue = 'HAS_VALUE',
+  inRange = 'IN_RANGE',
+  contains = 'CONTAINS',
+  dateComparison = 'DATE_COMPARISON',
+  dayOfYearInRange = 'DAY_OF_YEAR_IN_RANGE'
+}
```

Comment:
Huuuuge note here - not all of these operation types are supported by each path type, e.g. `NOT_EQUALS` is not supported by `USER_NAME` path, `DATE_COMPARISON` is not supported by `RELATION_PATH`, and so on. I can try to write some compatibility matrix if need.

Reply:
Yeah, I noticed this. Also sometimes the operation type will enforce a certain `queryValue.value` type. If creating that matrix wouldn't be too huge of a lift I think it would be quite valuable.